### PR TITLE
Update to even newer OMF plugin structure

### PR DESCRIPTION
**Pull request by sagebind**
Updates the package to the even _newer_ OMF plugin structure, utilizing function autoloading for startup speed.

Also now uses the core `available` utility to check for existing Composer commands, which also fixes junk output when it does not exist.

### CODE REVIEW

File: functions/composer.fish [hunk 1]
```diff
@@ -0,0 +1,10 @@
+function composer -d "Dependency manager for PHP"
+  if test -e ./composer.phar # if there is composer in this path use local one
+    ./composer.phar $argv;
+  else if test -n "$COMPOSER_BIN"
+    eval "$COMPOSER_BIN" $argv;
+  else
+    echo "Please install composer.phar to your PATH. Or use \"composer_install\" to install Composer to current directory."
```
Comment (derekstavis) [inline functions/composer.fish:7]: Now it's `composer.install`, but I think we should leave it named `composer_install`.

Comment (sagebind) [inline functions/composer.fish:7]: Okieday then. Will fix.

**Comment by sagebind** [comment]
Squashed the commits. Ping @derekstavis

**Comment by derekstavis** [comment]
A big 👍

**Comment by bpinto** [comment]
Thanks @coderstephen !

**Pull request merged**